### PR TITLE
feat(picker): support columnsCount

### LINKED ISSUE #1712: [Feature] support choosing city without region for picker component
# Feature request 🚀

### Expected:

- No breaking changes

### Examples:

```js

```

### PR DESCRIPTION
**Pull request by pureStyles**
## Checklist

List of tasks you have already done and plan to do.

- [x] Fix linting errors
- [x] Tests have been added / updated (or snapshots)

## Change information

support columnsNum prop for Picker(include multiple column mode and cascade column mode).

## Issues

close #1712

## Related Links

wiki link for the technical solution: [technical solution link](https://www.yuque.com/oriheart/tqbfcp/angcwsippwzlx9y3?singleDoc)

### CODE REVIEW

File: packages/varlet-ui/src/picker/Picker.vue [hunk 1]
```diff
@@ -149,6 +150,14 @@ export default defineComponent({
 
     let prevIndexes: number[] = []
 
+    watch(
```
Comment (chouchouji) [inline packages/varlet-ui/src/picker/Picker.vue:153]: maybe `computed` is a better choice

File: packages/varlet-ui/src/picker/props.ts [hunk 2]
```diff
@@ -47,6 +48,16 @@ export const props = {
     type: [Number, String],
     default: 6,
   },
+  columnsNum: {
+    type: [String, Number],
+    validator(value: string | number) {
```
Comment (chouchouji) [inline packages/varlet-ui/src/picker/props.ts:53]: no need this validator. You should process this prop in `computed` method

File: packages/varlet-ui/src/picker/docs/en-US.md [hunk 3]
```diff
@@ -447,6 +447,7 @@ function handleChange(values, indexes) {
 | `cascade` | Whether to enable cascade mode | _boolean_ | `true` |
 | `option-height` | The height of the option | _string \| number_ | `44` |
 | `option-count` | Number of visible options | _string \| number_ | `6` |
+| `columns-num`  | Number of visible columns | _string \| number_ | `Infinity` |
```
Comment (chouchouji) [inline packages/varlet-ui/src/picker/docs/en-US.md:450]: remember to add version `***3.3.11***`

**Comment by chouchouji** [comment]
Adding an example about this prop is useful to know how to use it

File: packages/varlet-ui/src/picker/Picker.vue (hunk 1 above)
Comment (pureStyles) [reply packages/varlet-ui/src/picker/Picker.vue:153]: okay~I agree.
I'm will make changes to this section of the code tonight, based on your advice.

File: packages/varlet-ui/src/picker/props.ts (hunk 2 above)
Comment (pureStyles) [reply packages/varlet-ui/src/picker/props.ts:53]:
```javascript
computed(() => {
if(...){
}
})
```

Comment (pureStyles) [reply packages/varlet-ui/src/picker/props.ts:53]:
```javascript
computed(() => {
    if(...){
        warn('Picker', '"columnsNum" should be > 0')
        return Infinity
    }
})
```
like this ?

**Comment by pureStyles** [comment]
> Adding an example about this prop is useful to know how to use it
get, i will add an example tonight🤡

File: packages/varlet-ui/src/picker/Picker.vue (hunk 1 above)
Comment (pureStyles) [reply packages/varlet-ui/src/picker/Picker.vue]: done-.-

File: packages/varlet-ui/src/picker/props.ts (hunk 2 above)
Comment (pureStyles) [reply packages/varlet-ui/src/picker/props.ts]: done-.-

File: packages/varlet-ui/src/picker/docs/en-US.md (hunk 3 above)
Comment (pureStyles) [reply packages/varlet-ui/src/picker/docs/en-US.md]: image

**Comment by pureStyles** [comment]
> Adding an example about this prop is useful to know how to use it

image

**Review by chouchouji — APPROVED** [review]

File: packages/varlet-ui/src/picker/Picker.vue [hunk 4]
```diff
@@ -174,7 +175,8 @@ export default defineComponent({
     }
 
     function normalizeNormalMode(columns: PickerColumnOption[][]) {
-      return columns.map((column, idx) => {
+      const visibleColumns = visibleColumnsCount.value ? columns.slice(0, visibleColumnsCount.value) : columns
```
Comment (haoziqaq) [inline packages/varlet-ui/src/picker/Picker.vue:178]: const visibleColumns = props.columnsCount != null ? columns.slice(0, visibleColumnsCount.value) : columns

File: packages/varlet-ui/src/picker/docs/zh-CN.md [hunk 5]
```diff
@@ -202,6 +202,32 @@ async function picker() {
 </template>
 ```
 
+### 配置显示列
+
+可以通过`columns-count`属性配置显示的列数，如果不设置或设置为`0`将显示所有列。
```
Comment (haoziqaq) [inline packages/varlet-ui/src/picker/docs/zh-CN.md:207]: When set to 0, it should not default to visible all columns.

**Review by haoziqaq — APPROVED** [review]

**Pull request merged**